content/applications/general/voip/onsip.rst

```diff
@@ -5,94 +5,180 @@ Use VoIP services in Odoo with OnSIP
+
+   .. image:: onsip/onsip04.png
       :align: center
+      :alt: Missing parameter message in the Odoo VoIP widget.
 
-You can now also receive phone calls. Your number is the one provided by OnSIP. Odoo will ring and
-display a notification.
+.. tip::
+   If an *Incorrect Number* message appears in the Odoo softphone, make sure to use the
+   international format for the number. This means leading with the :guilabel:`+ (plus)` sign
+   followed by the international country code.
 
-.. image:: onsip/onsip06.png
-   :align: center
+   A country code is a locator code that allows access to the desired country's phone system. The
+   country code is dialed first, prior to the target number. Each country in the world has its own
+   specific country code.\
 
-OnSIP on Your Cell Phone
-========================
+    E.g.: `+16506913277` (where `+1` is the international prefix for the United States).
 
-In order to make and receive phone calls when you are not in front of your computer, you can use a
-softphone app on your cell phone in parallel of Odoo VoIP. This is useful for on-the-go calls, but
-also to make sure to hear incoming calls, or simply for convenience. Any SIP softphone will work.
+   For a list of comprehensive country codes, visit: `https://countrycode.org
+   <https://countrycode.org>`_.
 
-On Android and iOS, OnSIP has been successfully tested with `Grandstream Wave
-<https://play.google.com/store/apps/details?id=com.grandstream.ucm>`_. When creating an account,
-select OnSIP in the list of carriers. You will then have to configure it as follows:
+   .. image:: onsip/onsip05.png
+      :align: center
+      :alt: Incorrect number message populated in the Odoo VoIP widget.
 
-- **Account name**: OnSIP
-- **SIP Server**: the OnSIP 'Domain'
-- **SIP User ID**: the OnSIP 'Username'
-- **SIP Authentication ID**: the OnSIP 'Auth Username'
-- **Password**: the OnSIP 'SIP Password'
+OnSIP on Your Cell Phone
+========================
 
-Aside from initiating calls from Grandstream Wave on your phone, you can also initiate calls by
-clicking phone numbers in your browser on your PC. This will make Grandstream Wave ring and route
-the call via your phone to the other party. This approach is useful to avoid wasting time dialing
-phone numbers. In order to do so, you will need the Chrome extension `OnSIP Call Assistant
+In order to make and receive phone calls when the user is not in front of Odoo on their computer, A
+softphone app on a mobile phone can be used in parallel with Odoo VoIP. This is useful for
+on-the-go calls, but also to make sure to hear incoming calls, or simply for convenience. Any SIP
+softphone will work.
+
+.. seealso::
+   :doc:`devices_integrations`
+
+Grandstream Wave
+----------------
+
+.. important::
+   A Grandstream Wave account is necessary to utilize the following integration. This includes a
+   networked PBX hardware integration from Grandstream (e.g. Grandstream UCM63XX series IP PBX).
```

Comment:
i do ont understand what you mean here. i need to have an account with access to specific hardware with grandstream? unclear to me